app/controllers/forms/what_happens_next_controller.rb

```diff
@@ -1,26 +1,52 @@
 module Forms
   class WhatHappensNextController < ApplicationController
     after_action :verify_authorized
+
     def new
       authorize current_form, :can_view_form?
       @what_happens_next_form = WhatHappensNextForm.new(form: current_form).assign_form_values
+      @preview_html = preview_html(@what_happens_next_form)
     end
 
     def create
       authorize current_form, :can_view_form?
       @what_happens_next_form = WhatHappensNextForm.new(**what_happens_next_form_params)
+      @preview_html = preview_html(@what_happens_next_form)
 
-      if @what_happens_next_form.submit
-        redirect_to form_path(@what_happens_next_form.form), success: t("banner.success.form.what_happens_next_saved")
-      else
-        render :new
+      case params[:route_to].to_sym
+      when :preview
+        if @what_happens_next_form.valid?
+          render :new, status: :ok
+        else
+          render :new, status: :unprocessable_entity
+        end
+      when :save_and_continue
+        if @what_happens_next_form.submit
+          redirect_to form_path(@what_happens_next_form.form), success: t("banner.success.form.what_happens_next_saved")
+        else
+          render :new, status: :unprocessable_entity
+        end
       end
     end
 
+    def render_preview
+      authorize current_form, :can_view_form?
```

Comment:
We need to add a request spec here or do we test this somewhere else? Eg. what happens if the user does not have permission to view this form. What do they see on their page and what actually happens?

Reply:
I don't think we're really doing this anywhere else and to me it would make sense for any tests around the authorisation behaviour to be in the application controller's tests or some separate authentication tests, since that's where the logic is, rather than add tests for this in every controller action. I think for this bit it should be enough to test that the controller is calling `Pundit.authorize` in each action, which I've now added.